src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetWrapperBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.OData;
+
+namespace Microsoft.AspNet.OData.Formatter.Deserialization
+{
+    /// <summary>
+    /// Encapsulates an <see cref="ODataResourceSet"/> and the <see cref="ODataResource"/>'s that are part of it.
+    /// </summary>        
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1012:AbstractTypesShouldNotHaveConstructors")]
+    public abstract class ODataResourceSetWrapperBase : ODataItemBase
+    {
+        /// <summary>
+        /// To determint the type of Resource Set
+        /// </summary>        
+        internal abstract ResourceSetType ResourceSetType { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ODataResourceSetWrapper"/>.
+        /// </summary>
+        /// <param name="item">The wrapped item.</param>
+        public ODataResourceSetWrapperBase(ODataResourceSetBase item)
+            : base(item)
+        {
+            Resources = new List<ODataResourceWrapper>();
+        }
+
+        /// <summary>
+        /// Gets the wrapped <see cref="ResourceSetBase"/>.
+        /// </summary>
+        public ODataResourceSetBase ResourceSetBase
```

Comment:
>public ODataResourceSetBase ResourceSetBase [](start = 8, length = 43)

ODataDeltaResourceSet

Reply:
Base is for DeltaResourceSet and ResourceSet right

---
In reply to: [604620220](https://github.com/OData/WebApi/pull/2455#discussion_r604620220) [](ancestors = 604620220)

Reply:
@Sreejithpin - Is this resolved?  This file is no longer in the PR.

---
In reply to: [604694595](https://github.com/OData/WebApi/pull/2455#discussion_r604694595) [](ancestors = 604694595,604620220)